src/framework/theme/components/stepper/stepper.component.ts

```diff
@@ -191,14 +206,20 @@ export class NbStepperComponent {
    * Navigate to next step
    * */
   next() {
-    this.selectedIndex = Math.min(this.selectedIndex + 1, this.steps.length - 1);
+    if (this.selectedIndex !== this.steps.length - 1) {
+      this.selectedIndex = Math.min(this.selectedIndex + 1, this.steps.length - 1);
+      this.stepChanged.emit(this.selectedIndex);
+    }
   }
 
   /**
    * Navigate to previous step
    * */
   previous() {
-    this.selectedIndex = Math.max(this.selectedIndex - 1, 0);
+    if (this.selectedIndex !== 0) {
+      this.selectedIndex = Math.max(this.selectedIndex - 1, 0);
```

Comment:
I don't think we need to use `Math.max` here, because we already have appropriate `if` above.